### PR TITLE
Navbar IE compatibility improvements and first pass at "Layout" section

### PR DESCRIPTION
**Pull request by jgolieb**
This is kind of a big PR because I got very backlogged while waiting to get the previous two PRs committed and initially my branching was a little sloppy. As such, I decided the best thing to do was merge two of my branches and PR them together. So this includes two main things:

1. IE compatibility improvements to the navbar.
2. Addition of the Layout section to the demo page, which shows the behavior of .container-fluid in this theme, and the accompanying JS to control the demo page.

On top of that, a number of values were replaced with variables and a number of !important flags were removed from the navbar. There are further tweaks to improve the source code, the project file structure, and the build system coming in additional PRs after this one, so if you have issues with things related to coding style or project organization, by all means let me know, but I may have already addressed it.

### CODE REVIEW

File: src/components/_type.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * modification, redistribution, sharing, lending or other exploitation
+ * of all or any part of the contents of this file is strictly prohibited.
+ */
+
+h1, h2, h3, h4, h5, h6,
+.h1, .h2, .h3, .h4, .h5, .h6 {
+  color: $text-dark;
+
+  &:last-child {
+    margin-bottom: 0;
```
Comment (mihai-mihalache) [inline src/components/_type.scss:14]: what's the point of this?

File: application.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2011-2018, Hortonworks Inc.  All rights reserved.
+ * Except as expressly permitted in a written agreement between you
+ * or your company and Hortonworks, Inc, any use, reproduction,
+ * modification, redistribution, sharing, lending or other exploitation
+ * of all or any part of the contents of this file is strictly prohibited.
+ */
+
+#containerExample,
+#containerExample,
```
Comment (mihai-mihalache) [inline application.scss]: I couldn't not notice this :)

Comment (mihai-mihalache) [reply application.scss]: I think it's some output of the build process but it has an underlying problem which should be solved.

File: src/components/_type.scss (hunk 1 above)
Comment (jgolieb) [reply src/components/_type.scss:14]: We want to maintain a consistent amount of padding around all content in Fluid (which is all supposed to be inside of panels) so I remove the bottom margin from the last child so it doesn't look like there's greater padding on the bottom.

File: application.scss (hunk 2 above)
Comment (jgolieb) [reply application.scss]: It's very likely that I just doubled that up when resolving a merge conflict. I'll fix.

**Review by mihai-mihalache — APPROVED** [review]

**Review by sardell — APPROVED** [review]
This all looks good to me. +1.

**Pull request merged**